bin/ScanPrep.py

```diff
@@ -0,0 +1,253 @@
+import zarr
+from skimage.filters import gaussian
+from skimage import measure
+import torch.nn as nn
+import numpy as np
+import pandas as pd
+from jsonargparse import CLI
+import matplotlib.pyplot as plt
+
+
+
+class GuasianTissueThresholder:
+    """
+    A module for identifying tissue using a guasian blur+threshold, 
+    handelded as a module so alternate options can be added in the future if desired.
+
+    Atributes
+    ---------------
+    thresh : float
+        value between 0 & 1 indicating the intensity treshold ove whice a pixel is considered white background
+    channels : tuple of ints
+        indicates which channels to include in the intensity calculation
+    agg_function : funtion
+        Takes a 5D array in [T,C,Z,Y,X] order (default ome ordering) as input and returns a 2D intensity array
+    sigma : float
+        sigma value for the gausian blur
+    
+    Methods
+    ----------
+    forward(arr)
+        Applies a gausian blur and treshold to detect tissue in a bright field image
+    """
+    def __init__(self,thresh,channels,agg_function,sigma=10):
+        """
+        Constructs module
+
+        Parameters
+        ---------------
+        thresh : float
+            value between 0 & 1 indicating the intensity treshold ove whice a pixel is considered white background
+        channels : tuple of ints
+            indicates which channels to include in the intensity calculation
+        agg_function : funtion
+            Takes a 5D array in [T,C,Z,Y,X] order (default ome ordering) as input and returns a 2D intensity array
+        sigma : float
+            sigma value for the gausian blur
+        """
+        self.thresh=thresh
+        self.channels=channels
+        self.agg_function=agg_function
+        self.sigma=sigma
+    def __call__(self,arr):
+        """
+        Applies a gausian blur and treshold to detect tissue in a bright field image
+
+        Parameters
+        ---------------
+        arr : array
+            5D array in default OME order (T,C,Z,Y,X)
+        
+        Returns
+        ---------------
+        Intensity map of arr[0,:,0,:,:] based on the module attributes  
+        """
+        arr_sub=np.copy(arr[0,self.channels,0,:,:])
```

Comment:
More of safety step, since I didn't know how to enforce the provided function returning a copy. It isn't realy required with the versions I implemented but long term I was hoping that it could be expanded to allow the user to provide more complex agg functions. there's some tools that let you approximate a stain intensity for an arbitrary stain. can remove for now if you think we should.